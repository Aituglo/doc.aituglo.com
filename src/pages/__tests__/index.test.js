/* eslint-env jest */
import * as React from 'react'
import { mount, shallow } from 'enzyme'
import { Root } from '../index'

const EDGES = [
  {
    node: {
      id: 'sup',
      context: {
        nodePath: '/vim',
        category: 'Apps',
        title: 'Vim'
      }
    }
  }
]

describe('Main page', () => {
  let wrap, swrap

  beforeEach(() => {
    const data = {
      siteSearchIndex: {
        index: {}
      },
      allPages: {
        edges: EDGES
      }
    }

    wrap = mount(<Root data={data} />)
    swrap = shallow(<Root data={data} />)
  })

  it.skip('works', () => {
    console.log(wrap.debug())
  })

  it('matches (shallow)', () => {
    expect(swrap).toMatchSnapshot()
  })

  it('matches (deep)', () => {
    expect(wrap).toMatchSnapshot()
  })

  it('has links to vim', () => {
    expect(wrap.containsMatchingElement(<code>vim</code>)).toBeTruthy()
    expect(
      wrap.containsMatchingElement(<span className='title'>Vim</span>)
    ).toBeTruthy()
  })

  it('headings', () => {
    expect(
      wrap
        .find('.pages-list h2')
        .at(0)
        .text()
    ).toEqual('Recently updated')
  })
})
