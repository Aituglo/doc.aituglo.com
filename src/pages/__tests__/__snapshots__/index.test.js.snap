// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page matches (deep) 1`] = `
<Root
  data={
    Object {
      "allPages": Object {
        "edges": Array [
          Object {
            "node": Object {
              "context": Object {
                "category": "Apps",
                "nodePath": "/vim",
                "title": "Vim",
              },
              "id": "sup",
            },
          },
        ],
      },
      "siteSearchIndex": Object {
        "index": Object {},
      },
    }
  }
>
  <StaticQuery
    render={[Function]}
  >
    <div>
      Loading (StaticQuery)
    </div>
  </StaticQuery>
</Root>
`;

exports[`Main page matches (shallow) 1`] = `
<StaticQuery
  render={[Function]}
/>
`;
